pkg/server/serverpb/admin.proto

```diff
@@ -455,6 +456,57 @@ message DecommissionStatusResponse {
   repeated Status status = 2 [(gogoproto.nullable) = false];
 }
 
+// TODO(irfansharif): Should all these definitions for the orchestrator process
+// sit in a new server entirely? They're pretty self contained. We've only
+// dumped them into the admin server out of laziness.
+
+// AckClusterVersionRequest is used to inform the target node of a cluster
+// version bump. The target node is responsible for durably persisting the
+// message and enabling the corresponding version gates.
+//
+// TODO(irfansharif): Implement custom stringer method for these.
+message AckClusterVersionRequest {
+  roachpb.Version version = 1;
+}
+
+// AckClusterVersionResponse is the response to an AckClusterVersionRequest.
+message AckClusterVersionResponse { }
+
+// An EveryNodeRequestUnion contains exactly one of the requests. The values
+// added here must match those in EveryNodeResponseUnion.
+message EveryNodeRequestUnion {
+  oneof value {
+    AckClusterVersionRequest ack_cluster_version = 1;
+  }
+}
+
+// An EveryNodeResponseUnion contains exactly one of the responses. The values
+// added here must match those in EveryNodeRequestUnion.
+message EveryNodeResponseUnion {
+  oneof value {
+    AckClusterVersionResponse ack_cluster_version = 1;
+  }
+}
+
+// An EveryNodeRequest contains a single request from the list defined in
+// EveryNodeRequestUnion. These requests will power the long running migrations
+// process where the orchestrator of these long running migrations may need to
+// run arbitrary operations across every node in the cluster (depending on the
+// specific migration being executed). One such example is the orchestrator
+// wanting to inform every node in the cluster that a certain cluster version is
+// now active, and that each node should enable the relevant version gates
+// corresponding to said version.
+//
+// TODO(irfansharif): Actually use these as outlined above.
+message EveryNodeRequest {
```

Comment:
As we briefly discussed, `EveryNode` is not a fitting name. That name somehow suggests that this RPC internally will do something on every node, but this is not how it works.

I also come down on the side of thinking that this should be a service and that an RPC can be added for each individual task. The argument that I've heard for the status quo in this PR, that it is somehow less work, does not convince me as you have also added custom codegen, so it's `make generate` vs `make protobuf`. At that point I would rather stick with the idiomatic choice:

```go
service Migrations {
  rpc BumpClusterVersion(BumpClusterVersionRequest) returns BumpClusterVersionResponse {}
  rpc CheckReplicaVersions(CheckReplicaVersionsRequest) returns CheckReplicaVersionsResponse {} // maybe needed for range migrations
  // ...
}
```

I also think this is a good work item to go into a separate PR. It's 100% clear that we will need it, and until the comments are just right etc I think we'll have 1-2 more review iterations that would clutter up this main PR.